src/components/plots/ViolinPlot.jsx

```diff
@@ -19,7 +20,8 @@ const ViolinPlot = (props) => {
   const dispatch = useDispatch();
 
   const geneExpression = useSelector((state) => state.genes.expression);
-  const cellSets = useSelector((state) => state.cellSets);
+  const cellSets = useSelector(getCellSets());
```

Comment:
I think `clustersAvailable` here is actually using something similar to the getCellSetsHierrchy I mentioned in other files, right? Would we be able to reuse it here too and perhaps refactor a bit the code here to adapt it?

Reply:
This checks if there are available hierarchy for that type. `getCellSetsHierarchy` returns the clusters inside `properties` which belong to the same parent. So its different.